### PR TITLE
Marketplace business services: Feature branch

### PR DESCRIPTION
**Pull request by KokkieH**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

WooCommerce.com will be introducing a new type of product, Business Services, to the marketplace. (Internal post: p9h1Lv-c9Z-p2)

This PR adds support for including this new product type in the in-app marketplace at WooCommerce->Extensions.

Closes https://github.com/Automattic/woocommerce.com/issues/20444

This PR contains the changes from:

|PR|Description|
|---|---|
|#46530|Adds a Business Services tab to the marketplace.|
|#46725|Displays the new product type and categories on the new tab.|
|#47048|Fetch and display featured Business Service products on the Discover tab.|
|#47135|Add Business Services to the Search tab.|
|#47183|Add a new card type for the Business Services product type.|
|#47227|Ensure the Business Services tab only appears if there are products of that type on WooCommerce.com|
|#47352|Remove heading on Business Services tab, and some other small tweaks|
|#47452|Hide notices on Business Services and My Subscriptions tabs, and refactor search results to simplify component|

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Prep

To test the request from your local WooCommerce core wp-env environment to your local WCCOM environment, you first need a running instance of WCCOM, as no products of the new type currently exists on production. 

1. Make sure your WCCOM environment is running on the latest version of the `feature/business-services-tab` branch.
2. Make sure you have a few products on the Business Services category, as well as its subcategories (see testing instructions for https://github.com/Automattic/woocommerce.com/pull/20275). 
3. Go to https://woocommerce.test/wp-admin/edit.php?post_type=featured_product and ensure that you have multiple featured products in different groups. Some of them should be in the Business Services group.
4. Check that the group is being returned from the featured endpoint https://woocommerce.test/wp-json/wccom-extensions/3.0/featured?country=US. (You'll probably need to flush the transient that caches this on your WCCOM with `wp cache flush`.)

Next you'll need to point your WC Core environment to your local WCCOM, as the `/wccom-extensions/3.0` API is not available yet on production:

1. Add a mu-plugin to your WC Core wp-env (as suggested in p6JqRr-aWb-p2), and put these two filters in it:

```php
add_filter( 'http_request_host_is_external', '__return_true' );

add_filter( 'http_request_args', function ( $r ) {
	$r['reject_unsafe_urls'] = false;
	$r['sslverify']          = false;
	$r['timeout']            = 60;

	return $r;
} );
```

2. Get the ID of wp-env's WordPress container.

image

3. Link the container to woocommerce.test in the Docker network.

```bash
docker network connect woocommerce.test <container ID>
```

#### Test

1. Check out this branch
2. Edit `plugins/woocommerce/includes/admin/class-wc-admin-addons.php` and replace the start of the `fetch_featured()` function with the following:

```php
	public static function fetch_featured() {
		$transient_name = 'wc_addons_featured';
		// Important: WCCOM Extensions API v3.0 is used.
		$url      = 'https://woocommerce.test/wp-json/wccom-extensions/3.0/featured';
		$locale   = get_user_locale();
		$featured = false;
```

3. In `plugins/woocommerce-admin/client/marketplace/components/constants.ts`, edit the `MARKETPLACE_HOST` const to be `https://woocommerce.test`
4. Go to `/plugins/woocommerce` and run `pnpm --filter='@woocommerce/plugin-woocommerce' build`
5. Visit the [in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions). Confirm a Business Services tab appears after a short delay. 
6. Confirm that a new section appears on the Discover tab containing the Business Service product you set as featured on WCCOM.
7. Switch to the Business Services tab. Confirm it shows your products, a list of Business Services child categories, and that you can filter the products by category.
8. Check the product cards against the designs: 5jyiYtoTF2ZUxgkulPVpOh-fi-3%3A3088
9. Open the dev console and run `localStorage.setItem( 'debug', 'wc-admin:*' );` Click a Business Service product card, then check that data for a Tracks event has been added to local storage (the dev env doesn't actually fire a Tracks event)
10. Use Search with a term that should return a Business Services product. If only Business Services are returned, you should see all matching products. If multiple product types are returned, they should be divided into plugin, theme and bs sections based on the product type.
11. Confirm that the other tabs also still look and work as expected.
12. Reverse the change made in step 3, and rebuild assets. Reload the marketplace. Now the Business Services tab should not appear (as you're querying production which doesn't have any of these products yet.

For more detailed instructions, please see the individual PRs.

Extensions_‹_WooCommerce_‹_woocommerce
Extensions_‹_WooCommerce_‹_woocommerce
Extensions_‹_WooCommerce_‹_woocommerce
Extensions_‹_WooCommerce_‹_woocommerce

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 

Add Business Services to the Extensions catalogue

#### Comment 

</details>

### CODE REVIEW

**Review by mcliwanow — APPROVED** [review]
I did another round of code review (not very thorough this time, as it was already reviewed in individual PRs) and tested the functionality with site connected to my local WooCommerce.test site.

Everything works as expected, considering that we'll merge https://github.com/woocommerce/woocommerce/pull/47452 too. 

It's ready to be shipped from my point of view.

File: plugins/woocommerce-admin/client/marketplace/README.md [hunk 1]
```diff
@@ -9,13 +9,14 @@ The page contains two parts, the WooCommerce.com marketplace and a list of produ
 - **Discover**: A curated list of extensions and themes.
 - **Browse**: All extensions.
 - **Themes**: All themes.
+- **Business Services**: All business services
```
Comment (Dan-Q) [inline plugins/woocommerce-admin/client/marketplace/README.md]: All the other items in this list end with a period.

```suggestion
- **Business Services**: All business services.
```

**Review by Dan-Q — APPROVED** [review]
Behaves as advertised; nice work. One non-blocking (one character) documentation suggestion.

**Comment by andfinally** [comment]
Just checking we're sizing the images as we intend – for themes, we do this on the container

```scss
overflow: hidden;
padding-top: 75%;
position: relative;
```

and this on the image

```scss
width: 100%;
position: absolute;
top: 0;
left: 0;
bottom: 0;
right: 0;
```

This constrains the image to 0.75 ratio while filling the container.

image

I'm sure there's a more modern way to do this, e.g. `object-fit: cover;` and `aspect-ratio: 0.75;`. But I'm wondering if it's intentional that the image shows up as contained in my browser.

image

**Comment by andfinally** [comment]
Our conditional loading of the tab means there's a slight delay before it renders. It's not a massive problem, and it's better than showing an empty tab, but it would be worth making the tab permanent some time after this is released.

https://github.com/woocommerce/woocommerce/assets/1647564/b9d91be2-8ee8-4120-b98b-6baf00725043

**Comment by KokkieH** [comment]
> Our conditional loading of the tab means there's a slight delay before it renders. It's not a massive problem, and it's better than showing an empty tab, but it would be worth making the tab permanent some time after this is released.

Yeah, this is unfortunately unavoidable atm--we can only decide whether or not to show the tab once we received search results from WCCOM, so either we only show it after a delay, or we show it and hide it after a delay.

Once we actually have products of this type on WCCOM we can definitely remove the selective rendering of this tab.

**Pull request merged**